lib/FreeRTOS/freertos-plus-mqtt/agent_command_pool.c

```diff
@@ -66,11 +66,19 @@ Command_t * Agent_GetCommand( uint32_t blockTimeMs )
 
     if( !initialized )
     {
-        memset( ( void * ) commandStructurePool, 0x00, sizeof( commandStructurePool ) );
-        freeCommandStructMutex = xSemaphoreCreateCounting( MQTT_COMMAND_CONTEXTS_POOL_SIZE, MQTT_COMMAND_CONTEXTS_POOL_SIZE );
-        configASSERT( freeCommandStructMutex ); /*_RB_ Create all objects here statically. */
+        taskENTER_CRITICAL();
+        {
+            if( !initialized )
+            {
+                memset( ( void * ) commandStructurePool, 0x00, sizeof( commandStructurePool ) );
+                freeCommandStructMutex = xSemaphoreCreateCounting( MQTT_COMMAND_CONTEXTS_POOL_SIZE, MQTT_COMMAND_CONTEXTS_POOL_SIZE );
```

Comment:
FreeRTOS API functions cannot be invoked with in a critical section refer to doc [here](https://www.freertos.org/taskENTER_CRITICAL_taskEXIT_CRITICAL.html). So this call needs to be outside of the critical section.